### ISSUE TITLE
Removing min_master_version for a cluster that previously had it set returns an error 

### ISSUE THREAD

**Issue by primeroz** [terraform-providers/terraform-provider-google#6418]
<!--- Please leave this line, it helps our automation: [issue-type:bug-report] --->
<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

<!--- Thank you for keeping this note for the community --->

### Terraform Version

Terraform: v0.12.23
Google ProvideR: 3.22 


### Affected Resource(s)

* google_container_cluster

### Overview

See https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/522 

When setting `min_master_version` to `null` , because of setting the `release_channel` , for a cluster that previously had it set causes an error when , i believe, the state is about to be updated.

In my case i had , for a long time, kept both the release_channel and the min_master_version in the config and always had to update the min_master_version to match the currently available one. 
Now i decided to remove it since apparently is optional.

I can see that the the `UPDATE_CLUSTER` operation is successful and GOOGLE API returns 200 , after that the error is returned by terraform 

```
 -----------------------------------------------------
 2020/05/19 13:37:58 [DEBUG] Google API Response Details:
 ---[ RESPONSE ]--------------------------------------
 HTTP/1.1 200 OK
 Connection: close

 {                                                                
  "name": "operation-1589895468012-1ef861ce",                                                                                                                                                                                                                                                               
  "zone": "europe-west1",                                                                                                                                                                                                                                                                                   
  "operationType": "UPDATE_CLUSTER",                                                                                                                                                                                                                                                                        
  "status": "DONE",                                                                                                                                                                                                                                                                                         
  "selfLink": "https://container.googleapis.com/v1beta1/projects/PROJECT/locations/europe-west1/operations/operation-1589895468012-1ef861ce",                                                                    
  "targetLink": "https://container.googleapis.com/v1beta1/projects/PROJECT/locations/europe-west1/clusters/CLUSTER",                                                                                                 
  "startTime": "2020-05-19T13:37:48.012147826Z",                  
  "endTime": "2020-05-19T13:37:48.389156153Z"                                                                                                                                                                                                                                                               
 }
                                                                  
                                                                  
 -----------------------------------------------------
 2020/05/19 13:37:58 [DEBUG] Retry Transport: Stopping retries, last request was successful
2020-05-19T13:37:58.471Z [DEBUG] plugin.terraform-provider-google-beta_v3.22.0_x5: 2020/05/19 13:37:58 [DEBUG] Retry Transport: Returning after 1 attempts
2020-05-19T13:37:58.471Z [DEBUG] plugin.terraform-provider-google-beta_v3.22.0_x5: 2020/05/19 13:37:58 [DEBUG] Got DONE while polling for operation operation-1589895468012-1ef861ce's status                                                          
2020-05-19T13:37:58.471Z [DEBUG] plugin.terraform-provider-google-beta_v3.22.0_x5: 2020/05/19 13:37:58 [DEBUG] Unlocking "google-container-cluster/PROJECT/europe-west1/CLUSTER"
2020-05-19T13:37:58.471Z [DEBUG] plugin.terraform-provider-google-beta_v3.22.0_x5: 2020/05/19 13:37:58 [DEBUG] Unlocked "google-container-cluster/PROJECT/europe-west1/CLUSTER"
2020-05-19T13:37:58.471Z [DEBUG] plugin.terraform-provider-google-beta_v3.22.0_x5: 2020/05/19 13:37:58 [INFO] GKE cluster fc1 master authorized networks config has been updated
2020/05/19 13:37:58 [DEBUG] module.cluster.google_container_cluster.primary: apply errored, but we're indicating that via the Error pointer rather than returning it: Malformed version: 
2020/05/19 13:37:58 [ERROR] module.cluster: eval: *terraform.EvalApplyPost, err: Malformed version: 
2020/05/19 13:37:58 [ERROR] module.cluster: eval: *terraform.EvalSequence, err: Malformed version: 
```

### Expected Behavior

Update is successful and the `min_master_version` is removed from the state 

### Actual Behavior

Terraform return an error after the cluster update operation is completed

### Steps to Reproduce

1 .Create a GKE Cluster with both min_master_version and release_channel set
1. apply
1. remove the min_master_version or set it to null ( same effect ) 
1. apply
1. get the error

### References
* https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/522

**Comment by primeroz** [comment]
as a test i did 

* terraform state rm CLUSTER
* terraform import CLUSTER PROJECT/REGION/CLUSTER

and now the min_master_version in the state is indeed `null`, terraform apply now works and even after apply

```
$ terraform state pull | grep min_master_version
            "min_master_version": null,
```

It does seem to be an issue only when Changing from a `version` to `null` in the state

**Comment by venkykuberan** [comment]
@primeroz can you attach the full debug log here

**Comment by primeroz** [comment]
@venkykuberan I will try but I will need to build a custom project and make sure there are no sensitive information in the output so can't really tell when I ll be able to do so, also because of how verbose the debug output is and how long it will take me to go through it.

**Comment by slevenick** [comment]
@primeroz don't worry about providing the full debug logs, I'm able to reproduce this

**Comment by slevenick** [comment]
Looks like this was already fixed via https://github.com/terraform-providers/terraform-provider-google-beta/pull/2068

This is expected to be available in the 3.23.0 release of the provider next week.
